lib/daemon.js

```diff
@@ -62,6 +64,16 @@ class Daemon {
    */
   async start () {
     for await (const event of this._externalSecretEvents) {
+      // Check if the externalSecret should be managed by this instance.
+      if (event.object.spec) {
+        const externalSecretMetadata = event.object.metadata
+        const externalSecretController = event.object.spec.controllerID
+        if (externalSecretController && this._instanceId !== externalSecretController) {
+          this._logger.debug('the secret %s/%s is not managed by this instance but by %s',
+            externalSecretMetadata.namespace, externalSecretMetadata.name, externalSecretController)
+        }
```

Comment:
Looks like we are missing some skipping logic, now it would just log but then handle it anyways?

```suggestion
           continue
        }
```

or am I missing something here?